src/compiler/types.ts

```diff
@@ -2365,6 +2365,11 @@ namespace ts {
         sourceFiles: SourceFile[];
     }
 
+    export interface JsonSourceFile extends SourceFile {
+        jsonObject?: ObjectLiteralExpression;
+        extendedSourceFiles?: string[];

```

Comment:
Nit: Probably easier to make this non-optional and initialize to an empty array.